### PR TITLE
docs(metadata): update metadata swagger API docs for device profile changes

### LINKED ISSUE #3918: docs(metadata): Add new APIs for device profile changes
# 📚 Docs or Wiki Bug Report

### Description [**REQUIRED**]

 
Adding new APIs based on https://github.com/edgexfoundry/edgex-docs/pull/605
- Profile General Property PATCH API 
- Profile Device Resource POST API
- Profile Device Resource PATCH API 
- Profile Device Resource DELETE API 
- Profile Device Command POST API
- Profile Device Command PATCH API 
- Profile Device Command DELETE API

### What's the affected URL? [**REQUIRED**]
 
https://app.swaggerhub.com/apis-docs/EdgeXFoundry1/core-metadata/

### PR DESCRIPTION
**Pull request by jinlinGuan**
close #3918

Signed-off-by: Ginny Guan <ginny@iotechsys.com>

**If your build fails**  due to your commit message not passing the build checks, please review the guidelines here: https://github.com/edgexfoundry/edgex-go/blob/main/.github/Contributing.md

## PR Checklist
Please check if your PR fulfills the following requirements:

- [x] I am not introducing a breaking change (if you are, flag in conventional commit message with `BREAKING CHANGE:` describing the break)
- [x] I am not introducing a new dependency (add notes below if you are)
- [ ] I have added unit tests for the new feature or bug fix (if not, why?)NA, only update API docs
- [ ] I have fully tested (add details below) this the new feature or bug fix (if not, why?)NA, only update API docs
- [ ] I have opened a PR for the related docs change (if not, why?)NA, only update API docs
  <link to docs PR>

### CODE REVIEW

File: openapi/v2/core-metadata.yaml [hunk 1]
```diff
@@ -2227,6 +2277,421 @@ paths:
...
+      summary: "Allows basic information updates to an existing device profile"
+      requestBody:
+        required: true
+        content:
+          application/json:
+            schema:
+              type: array
+              items:
+                $ref: '#/components/schemas/DeviceProfileBasicInfoRequest'
+            example:
```
Comment (cloudxxx8) [inline openapi/v2/core-metadata.yaml:2293]: provide the example only modifying partial fields

File: openapi/v2/core-metadata.yaml [hunk 2]
```diff
@@ -2227,6 +2277,421 @@ paths:
...
+              $ref: '#/components/headers/correlatedResponseHeader'
+          content:
+            application/json:
+              schema:
+                $ref: '#/components/schemas/ErrorResponse'
+              examples:
+                500Example:
+                  $ref: '#/components/examples/500Example'
+    patch:
+      summary: "Allows device command updates to an existing device service"
```
Comment (cloudxxx8) [inline openapi/v2/core-metadata.yaml:2399]: not device service, and add the restriction statement

File: openapi/v2/core-metadata.yaml [hunk 3]
```diff
@@ -2227,6 +2277,421 @@ paths:
...
+        required: true
+        content:
+          application/json:
+            schema:
+              type: array
+              items:
+                $ref: '#/components/schemas/DeviceCommandRequest'
+            examples:
+              AddDeviceCommandRequest:
+                $ref: '#/components/examples/AddDeviceCommandRequest'
```
Comment (cloudxxx8) [inline openapi/v2/core-metadata.yaml:2410]: provide the example only modifying partial fields

File: openapi/v2/core-metadata.yaml [hunk 4]
```diff
@@ -2227,6 +2277,421 @@ paths:
...
+              $ref: '#/components/headers/correlatedResponseHeader'
+          content:
+            application/json:
+              schema:
+                $ref: '#/components/schemas/ErrorResponse'
+              examples:
+                500Example:
+                  $ref: '#/components/examples/500Example'
+    patch:
+      summary: "Allows updates to an existing device service"
```
Comment (cloudxxx8) [inline openapi/v2/core-metadata.yaml:2576]: not device service

File: openapi/v2/core-metadata.yaml [hunk 5]
```diff
@@ -2227,6 +2277,421 @@ paths:
...
+        required: true
+        content:
+          application/json:
+            schema:
+              type: array
+              items:
+                $ref: '#/components/schemas/DeviceResourceRequest'
+            examples:
+              AddDeviceResourceRequest:
+                $ref: '#/components/examples/AddDeviceResourceRequest'
```
Comment (cloudxxx8) [inline openapi/v2/core-metadata.yaml:2587]: provide the example only modifying partial fields

**Review by cloudxxx8 — CHANGES_REQUESTED** [review]

File: openapi/v2/core-metadata.yaml [hunk 6]
```diff
@@ -1194,6 +1220,30 @@ components:
...
+            name: "Switch"
+            readWrite: "RW"
+            resourceOperations:
+              - deviceResource: "SwitchButton"
+                DefaultValue: "false"
+    AddDeviceResourceRequest:
+      value:
+        - requestId: "e6e8a2f4-eb14-4649-9e2b-175247911369"
+          apiVersion: v2
+          deviceName: "Simple-Device-Profile"
```
[the remark this answers is not in the document]
Comment (jinlinGuan) [reply openapi/v2/core-metadata.yaml]: fixed

File: openapi/v2/core-metadata.yaml [hunk 7]
```diff
@@ -1194,6 +1220,30 @@ components:
                 resourceOperations:
                   - deviceResource: "SwitchButton"
                     DefaultValue: "false"
+    AddDeviceCommandRequest:
+      value:
+        - requestId: "e6e8a2f4-eb14-4649-9e2b-175247911369"
+          apiVersion: v2
+          deviceProfile: "Simple-Device-Profile"
```
[the remark this answers is not in the document]
Comment (jinlinGuan) [reply openapi/v2/core-metadata.yaml]: fixed

File: openapi/v2/core-metadata.yaml (hunk 1 above)
Comment (jinlinGuan) [reply openapi/v2/core-metadata.yaml:2293]: updated

File: openapi/v2/core-metadata.yaml (hunk 2 above)
Comment (jinlinGuan) [reply openapi/v2/core-metadata.yaml]: updated

File: openapi/v2/core-metadata.yaml (hunk 3 above)
Comment (jinlinGuan) [reply openapi/v2/core-metadata.yaml]: updated

File: openapi/v2/core-metadata.yaml (hunk 4 above)
Comment (jinlinGuan) [reply openapi/v2/core-metadata.yaml]: fixed

File: openapi/v2/core-metadata.yaml (hunk 5 above)
Comment (jinlinGuan) [reply openapi/v2/core-metadata.yaml]: updated

**Review by cloudxxx8 — APPROVED** [review]
LGTM

**Review by lenny-intel — APPROVED** [review]
LGTM

**Pull request merged**